clients/ranchercli/client.go

```diff
@@ -0,0 +1,112 @@
+package ranchercli
+
+import (
+	"fmt"
+	osys "os"
+	"os/exec"
+	"strings"
+
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	rancher   = "./rancher"
+	createCmd = "create"
+	deleteCmd = "delete"
+	moveCmd   = "move"
+)
+
+type Client struct {
+	Session *session.Session
+}
+
+func (c *Client) Create(name string, args ...string) error {
+	args = append([]string{name, createCmd}, args...)
+	c.ExecuteCommand(rancher, args...)
+
+	c.Session.RegisterCleanupFunc(func() error {
+		return c.Delete(name, args...)
+	})
+
+	return nil
+}
+
+func (c *Client) Delete(name string, args ...string) error {
+	args = append([]string{name, deleteCmd}, args...)
+	c.ExecuteCommand(rancher, args...)
+
+	return nil
```

Comment:
Markus you have to do `return c.ExecuteCommand(rancher, args...)` we're completely ignoring the error returned.

Reply:
I'll fix that.

Reply:
Resolved.